ksql-engine/src/test/java/io/confluent/ksql/integration/IntegrationTestHarness.java

```diff
@@ -198,30 +208,40 @@ public void stop() {
   }
 
   public Map<String, RecordMetadata> publishTestData(String topicName, TestDataProvider dataProvider, Long timestamp) throws InterruptedException, ExecutionException, TimeoutException {
+    return publishTestData(topicName, dataProvider, timestamp, DataSource.DataSourceSerDe.JSON);
+  }
+
+  public Map<String, RecordMetadata> publishTestData(String topicName,
+                                                     TestDataProvider dataProvider,
+                                                     Long timestamp,
+                                                     DataSource.DataSourceSerDe dataSourceSerDe) throws InterruptedException, ExecutionException, TimeoutException {
     createTopic(topicName);
-    return produceData(topicName, dataProvider.data(), getSerializer(dataProvider.schema()), timestamp);
-  }
-
-  private Serializer getSerializer(Schema schema) {
-    if (dataFormat.equals(DataSource.DELIMITED_SERDE_NAME)) {
-      return new KsqlDelimitedSerializer(schema);
-    } else if (dataFormat.equals(DataSource.AVRO_SERDE_NAME)) {
-      return new KsqlGenericRowAvroSerializer
-          (schema, this.schemaRegistryClient, new
-               KsqlConfig(Collections.emptyMap()));
-    } else {
-      return new KsqlJsonSerializer(schema);
-    }
+    return produceData(topicName, dataProvider.data(), getSerializer(dataProvider.schema(), dataSourceSerDe), timestamp);
   }
 
-  private Deserializer<GenericRow> getDeserializer(Schema schema) {
-    if (dataFormat.equals(DataSource.DELIMITED_SERDE_NAME)) {
-      return new KsqlDelimitedDeserializer(schema);
-    } else if (dataFormat.equals(DataSource.AVRO_SERDE_NAME)) {
-      return new KsqlGenericRowAvroDeserializer(schema, this.schemaRegistryClient, false);
-    } else {
-      return new KsqlJsonDeserializer(schema);
+  private Serializer getSerializer(Schema schema, DataSource.DataSourceSerDe dataSourceSerDe) {
```

Comment:
These two methods are another case of where we could have a more OO approach so we don't need to switch on the same thing in multiple places...

Reply:
Yes, the serde classes and DataSourceSerDe are used in other places too. This would need more changes that we can do in a separate PR.

Reply:
Sure - can you raise an issue for it so we can make sure it gets done

Reply:
Created an issue for this: https://github.com/confluentinc/ksql/issues/688